src/Core/Product/AbstractProductPresenter.php

```diff
@@ -64,8 +64,8 @@ private function fillImages(
         }
 
         if (!isset($presentedProduct['cover'])) {
-            if (isset($presentedProduct['images'][0])) {
-                $presentedProduct['cover'] = $presentedProduct['images'][0];
+            if (count($presentedProduct['images']) > 0) {
+                $presentedProduct['cover'] = array_values($presentedProduct['images'])[0];
```

Comment:
I dont want to rely on an array key, this way we have the first item, regardless the key of table.